test/test_pr_resolution.sh

```
@@ -0,0 +1,249 @@
+}
+
+# setup_mocks_api_fail creates mock environment and places a `gh` mock that immediately exits with status 1 to simulate API failures.
+setup_mocks_api_fail() {
+  setup_mocks_base
+  printf '#!/usr/bin/env bash\nexit 1\n' > "$mock_dir/gh"
+  chmod +x "$mock_dir/gh"
+}
+
+# run_script executes the target script under test with the mock directory prepended to PATH, forwarding all arguments.
+run_script() {
+  PATH="$mock_dir:$ORIGINAL_PATH" bash "$script" "$@"
```

Shellcheck (0.11.0)

[warning] script is referenced but not assigned.

(SC2154)